traffic_ops/traffic_ops_golang/ats/parentdotconfig.go

```diff
@@ -0,0 +1,1393 @@
+				nullParentInfo, seen = removeStrDuplicates(nullParentInfo, seen)
+
+				// If the ats version supports it and the algorithm is consistent hash, put secondary and non-primary parents into secondary parent group.
+				// This will ensure that secondary and tertiary parents will be unused unless all hosts in the primary group are unavailable.
+
+				parents := ""
+
+				if atsMajorVer >= 6 && ds.MSOAlgorithm == "consistent_hash" && (len(secondaryParentInfo) > 0 || len(nullParentInfo) > 0) {
+					parents = `parent="` + strings.Join(parentInfo, "") + `" secondary_parent="` + strings.Join(secondaryParentInfo, "") + strings.Join(nullParentInfo, "") + `"`
+				} else {
+					parents = `parent="` + strings.Join(parentInfo, "") + strings.Join(secondaryParentInfo, "") + strings.Join(nullParentInfo, "") + `"`
+				}
+				textLine += parents + ` round_robin=` + ds.MSOAlgorithm + ` qstring=` + parentQStr + ` go_direct=false parent_is_proxy=false`
+
+				parentRetry := ds.MSOParentRetry
+				if atsMajorVer >= 6 && parentRetry != "" {
+					if unavailableServerRetryResponsesValid(ds.MSOUnavailableServerRetryResponses) {
+						log.Errorf("DEBUG unavailableServerRetryResponsesValid '%+v'\n", ds.MSOUnavailableServerRetryResponses)
+						textLine += ` parent_retry=` + parentRetry + ` unavailable_server_retry_responses=` + ds.MSOUnavailableServerRetryResponses
+					} else {
+						log.Errorf("DEBUG unavailableServerRetryResponsesValid '%+v' NOT\n", ds.MSOUnavailableServerRetryResponses)
+						if ds.MSOUnavailableServerRetryResponses != "" {
+							log.Errorln("Malformed unavailable_server_retry_responses parameter '" + ds.MSOUnavailableServerRetryResponses + "', not using!")
+						}
+						textLine += ` parent_retry=` + parentRetry
+					}
+					textLine += ` max_simple_retries=` + ds.MSOMaxSimpleRetries + ` max_unavailable_server_retries=` + ds.MSOMaxUnavailableServerRetries
+				}
+				textLine += "\n"
+				textArr = append(textArr, textLine)
+			} else {
+				// log.Errorf("DEBUG PCGT ds '%+v' neither origin shield nor multisite\n", ds.Name)
+			}
+		}
+		sort.Sort(sort.StringSlice(textArr))
+		text = hdr + strings.Join(textArr, "")
+	} else {
+		log.Errorf("DEBUG PCGen not top level\n")
+		// not a top level cache
+		data, err := getParentConfigDS(inf.Tx.Tx, serverInfo.ID) // TODO rename
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("Getting parent config DS data (non-top-level): "+err.Error()))
+			return
+		}
+
+		log.Errorf("DEBUG PCGen got data len %+v\n", len(data))
+
+		parentInfos, err := getParentInfo(inf.Tx.Tx, serverInfo)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("Getting server parent info (non-top-level: "+err.Error()))
+			return
+		}
+
+		log.Errorf("DEBUG PCGen got parent info len %+v\n", len(parentInfos))
+
+		done := map[string]tc.DeliveryServiceName{}                                       // map[originHost]ds
```

Comment:
the comments seem unnecessary

Reply:
This comment specifically explains what the `string` is in `map[string]`. I'd prefer to keep it, to make the code easier to understand.

Reply:
I guess I should say, the comment doesn't really explain what the map is even used for. Why not `originsToDSNames` instead of `done`. Then the comment is unnecessary and doesn't have to explain what the map holds.

Reply:
Done.